fix: trace-usage.service.test.ts failing on CI

## Summary

Unit test failing on CI in `src/server/traces/__tests__/trace-usage.service.test.ts`.

## Failing Test

```
TraceUsageService > checkLimit > when self-hosted (IS_SAAS=false) > returns exceeded: false without checking limits
```

## Error

```
AssertionError: expected "vi.fn()" to not be called at all, but actually been called 1 times

Received: 
  1st vi.fn() call:
    Array [
      "team-123",
    ]

❯ src/server/traces/__tests__/trace-usage.service.test.ts:121:15
```

The test expects `mockOrganizationRepository.getOrganizationIdByTeamId` to not be called when `IS_SAAS=false`, but it is being called.

## CI Link

https://github.com/langwatch/langwatch/actions/runs/21482529897/job/61882550925

## Notes

This was discovered during PR #1218 but is unrelated to that PR - appears to be a pre-existing issue on main.

--- a/langwatch/src/server/traces/trace-usage.service.ts
+++ b/langwatch/src/server/traces/trace-usage.service.ts
@@ -1,5 +1,4 @@
 import type { QueryDslBoolQuery } from "@elastic/elasticsearch/lib/api/types";
-import { FREE_PLAN } from "../../../ee/licensing/constants";
 import type { PrismaClient } from "@prisma/client";
 import { env } from "~/env.mjs";
 import { dependencies } from "~/injection/dependencies.server";
@@ -56,6 +55,11 @@ export class TraceUsageService {
     maxMessagesPerMonth?: number;
     planName?: string;
   }> {
+    // Self-hosted = unlimited traces, skip all limit checks
+    if (!env.IS_SAAS) {
+      return { exceeded: false };
+    }
+
     const organizationId =
       await this.organizationRepository.getOrganizationIdByTeamId(teamId);
     if (!organizationId) {
@@ -66,11 +70,6 @@ export class TraceUsageService {
       this.getCurrentMonthCount({ organizationId }),
       this.subscriptionHandler.getActivePlan(organizationId),
     ]);
-
-    // Self-hosted = unlimited traces
-    if (!env.IS_SAAS && plan === FREE_PLAN) {
-      return { exceeded: false };
-    }
 
     if (count >= plan.maxMessagesPerMonth) {
       return {

